Resources/Private/Partials/Application/Field/Language.html

```diff
@@ -62,6 +62,15 @@
 
 					</f:for>
 
+					<f:comment>Add one language form if languages are mandatory and no languages are given yet</f:comment>
+					<f:if condition="{settings.validation.languageSkills}">
+						<f:if condition="{application.languageSkills}">
+							<f:else>
+								<f:render section="languagerow" arguments="{languageID : 998}" />
```

Comment:
Maybe move this magic number to a setting with a readable name? Or at least assign a variable from the controller for this.

Reply:
@mbrodala 
This is not a "magic number", it is simply a number high enough to not collide with exisiting entries coming from the controller. No need to ever change this value, so why should this be a setting / controller variable?

Reply:
Exactly because of this: it's unclear just from reading. We need to keep in mind that our code should be easy to read and understand for 3rd party.

Reply:
Agreed, but moving the number into the controller or a setting would imply there is more logic behind it than it actually holds. Plus, if you ever worked with dynamic fields in Fluid you know why this is necessary.

(And yes, the LanguageSkills partial is a monstrum and needs refactoring to make things more readable overall... But this is a different topic.)

Reply:
Actually right now it looks like there is more logic to this due to the magicness of the numbers. At least an `<f:alias>` map could be used to give them some speaking names like `addIndex` (999) and `placeholderIndex`(998).

Reply:
Also, renamed languageID to languageOffset as it is not an actual ID. Should make things even more clear.